pypeit/reduce.py

```diff
@@ -0,0 +1,719 @@
+    # being updated?
+    def local_skysub_extract(self, sciimg, sciivar, tilts, waveimg, global_sky, rn2img, sobjs,
+                             model_noise=True, min_snr=2.0, std = False, fit_fwhm=False,
+                             maskslits=None, show_profile=False, show_resids=False, show_fwhm=False, show=False):
+        """
+        Perform local sky subtraction, profile fitting, and optimal extraction slit by slit
+
+        Wrapper to skysub.local_skysub_extract
+
+        Parameters
+        ----------
+        sobjs: object
+           Specobjs object containing Specobj objects containing information about objects found.
+        waveimg: ndarray, shape (nspec, nspat)
+           Wavelength map
+
+        Optional Parameters
+        -------------------
+
+        Returns:
+            global_sky: (numpy.ndarray) image of the the global sky model
+        """
+
+        self.sciimg = sciimg
+        self.sciivar = sciivar
+        self.tilts = tilts
+        self.waveimg = waveimg
+        self.global_sky = global_sky
+        self.rn2img = rn2img
+        order_vec = self.spectrograph.order_vec()
+        self.skymodel, self.objmodel, self.ivarmodel, self.outmask, self.sobjs = skysub.ech_local_skysub_extract(
+            self.sciimg, self.sciivar, self.mask, self.tilts, self.waveimg, self.global_sky,
+            self.rn2img, self.tslits_dict, sobjs, order_vec,
+            std=std, fit_fwhm=fit_fwhm, min_snr=min_snr, bsp = self.par['bspline_spacing'],
+            sn_gauss=self.par['sn_gauss'], model_full_slit=self.par['model_full_slit'], model_noise=model_noise,
+            show_profile=show_profile, show_resids=show_resids, show_fwhm=show_fwhm)
+
+
+        # Step
+        self.steps.append(inspect.stack()[0][3])
+
+        if show:
+            self.show('ech_local', sobjs = self.sobjs, slits= True)
+            self.show('ech_resid', sobjs = self.sobjs, slits= True)
+
+        return self.skymodel, self.objmodel, self.ivarmodel, self.outmask, self.sobjs
+
+
+
+def instantiate_me(spectrograph, tslits_dict, mask, **kwargs):
+    """
+    Instantiate the Reduce subclass appropriate for the provided
+    spectrograph.
+
+    The class must be subclassed from Reduce.  See :class:`Reduce` for
+    the description of the valid keyword arguments.
+
+    Args:
+        spectrograph
+            (:class:`pypeit.spectrographs.spectrograph.Spectrograph`):
+            The instrument used to collect the data to be reduced.
+
+        tslits_dict: dict
+            dictionary containing slit/order boundary information
+
+    Returns:
+        :class:`PypeIt`: One of the classes with :class:`PypeIt` as its
+        base.
+    """
+    indx = [ c.__name__ == spectrograph.pypeline for c in Reduce.__subclasses__() ]
+    if not np.any(indx):
+        msgs.error('Pipeline {0} is not defined!'.format(spectrograph.pypeline))
+    return Reduce.__subclasses__()[np.where(indx)[0][0]](spectrograph, tslits_dict, mask, **kwargs)
+
+
```

Comment:
Did I miss the Echelle child?
Was expecting one

Reply:
You did indeed miss it. It is on line 604.